Realm.Shared/Dynamic/MetaRealmObject.cs

```diff
@@ -50,64 +51,63 @@ public override DynamicMetaObject BindGetMember(GetMemberBinder binder)
             var arguments = new List<Expression>
             {
                 WeakConstant(_metadata.Table),
-                Expression.Field(GetLimitedSelf(), RealmObjectRowHandleField),
-                Expression.Constant(_metadata.ColumnIndices[property.Name])
+                Expression.Constant(_metadata.ColumnIndices[property.Name]),
```

Comment:
This needs to build an expression tree that describes a call to any of the `Get*Value/Set*Value` methods and because the parameters in those methods changed, the list of arguments in the expression needs to change as well.